coc/http.py

```diff
@@ -508,7 +510,7 @@ async def initialise_keys(self):
 
             resp_payload = await resp.json()
             ip = json_loads(base64_b64decode(resp_payload["temporaryAPIToken"].split(".")[1] + "====").decode("utf-8"))["limits"][1]["cidrs"][0].split("/")[0]
-
+            ip = self.ip or ip
```

Comment:
L512 should be behind a `if not self.ip` to avoid unnecessary loads, decodes etc